src/bag/core.py

```diff
@@ -665,8 +665,84 @@ def generate_cell(self, specs: Dict[str, Any],
 
         return final_netlist
 
-    def extract_cell(self, lib_name: str, cell_name: str, extract_type: Optional[str], extract_corner: Optional[str],
-                     netlist: str) -> None:
+    def lvs_cell(self, lib_name: str, cell_name: str) -> None:
+        """Run layout-vs-schematic (LVS) on a cell. Exports netlist and GDS from OA views.
+        
+        Parameters
+        ----------
+        lib_name : str
+            Library name
+        cell_name : str
+            Cell name
+        
+        """
+        print('running LVS...')
+        lvs_passed, lvs_log = self.run_lvs(lib_name, cell_name, run_rcx=True)
+        if lvs_passed:
+            print('LVS passed!')
+        else:
+            raise ValueError(f'LVS failed... log file: {lvs_log}')
+    
+    def lvs_cell_raw(self, cell_name: str, layout: str, netlist: str) -> None:
+        """Run layout-vs-schematic (LVS) on a cell given the netlist and GDS path.
+        
+        Parameters
+        ----------
+        cell_name : str
+            Cell name
+        layout : str
+            GDS file path
+        netlist : str
+            Netlist file path
+        
+        """
+        print('running LVS...')
+        lvs_passed, lvs_log = self.run_lvs("", cell_name, layout=layout, netlist=netlist)
+        if lvs_passed:
+            print('LVS passed!')
+        else:
+            raise ValueError(f'LVS failed... log file: {lvs_log}')
+    
+    def lvl_cell(self, gds_file: str, ref_file: str) -> None:
+        """Run layout-vs-layout (LVL) on two GDS files.
+        
+        Parameters
+        ----------
+        gds_file : str
+            GDS file path
+        ref_file : str
+            Reference GDS file path
+        
+        """
+        print('running LVL...')
+        lvl_passed, lvl_log = self.run_lvl(gds_file, ref_file)
+        if lvl_passed:
+            print('LVL passed!')
+        else:
+            raise ValueError(f'LVL failed... log file: {lvl_log}')
+    
+    def nvn_cell(self, cell_name: str, netlist: str, ref_file: str) -> None:
+        """Run netlist-vs-netlist (NVN) on two netlist files.
```

Comment:
Slight nitpicking on the method names: Ideally these should be the names and implementations:
LVL: should take layout as input and stream out gds for comparing
GVG: should take gds as input
SVS: should take schematic as input and stream out netlist for comparing
NVN: should take netlist as input

So the NVN method is correctly named, but the LVL should be GVG. Alternatively you could do something similar to `lvs_cell()` and `lvs_cell_raw()`. But it's ok to keep the names as-is for now.